src/main/java/org/spongepowered/mod/mixin/core/event/entity/MixinEventEntityExplosion.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.mod.mixin.core.event.entity;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Predicate;
+import net.minecraft.util.BlockPos;
+import net.minecraft.world.Explosion;
+import net.minecraftforge.event.world.ExplosionEvent;
+import org.spongepowered.api.block.BlockLoc;
+import org.spongepowered.api.entity.Entity;
+import org.spongepowered.api.event.entity.EntityExplosionEvent;
+import org.spongepowered.api.util.annotation.NonnullByDefault;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+import org.spongepowered.api.world.extent.Extent;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.mod.interfaces.IMixinExplosion;
+import org.spongepowered.mod.util.VecHelper;
+import org.spongepowered.mod.wrapper.BlockWrapper;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+@NonnullByDefault
+@Mixin(value = ExplosionEvent.Detonate.class, remap = false)
+public abstract class MixinEventEntityExplosion extends ExplosionEvent implements EntityExplosionEvent {
+
+    @Shadow
+    public abstract List<BlockPos> getAffectedBlocks();
+
+    private List<BlockLoc> affectedBlocks;
+
+    public MixinEventEntityExplosion(net.minecraft.world.World world, Explosion explosion) {
+        super(world, explosion);
+        this.affectedBlocks = new ArrayList<BlockLoc>();
```

Comment:
`MixinBlockUpdateEvent` is actually incorrect, none of the instructions are added to the target class. (Can be seen by looking at the output of the mixin transformer).

In this case, you can make an injection to the constructor, `@Inject(method = "<init>, at = @At("RETURN"))`
and add the blocks there